pkg/block/azure/client_cache.go

```diff
@@ -28,35 +30,27 @@ func NewCache(p params.Azure) *ClientContainerCache {
 	}
 }
 
-// storageAccountRegex is used to extract the storage account from azure storage url.
-// Following rules from https://learn.microsoft.com/en-us/azure/storage/common/storage-account-overview#storage-account-name
-var storageAccountRegex = regexp.MustCompile("https://([a-z0-9]{3,24}).blob.core.windows.net/.*")
-
-func ExtractStorageAccount(storageAccountURL string) (string, error) {
-	matches := storageAccountRegex.FindAllStringSubmatch(storageAccountURL, -1)
-	if len(matches) != 1 || len(matches[0]) != 2 {
-		return "", ErrUnknownAzureURL
+func ExtractStorageAccount(storageAccount *url.URL) (string, error) {
```

Comment:
Can we move it to the adapter module - as this is not directly use in this module